cmd/dagger/functions.go

```diff
@@ -394,20 +394,19 @@ func (fc *FuncCommand) initializeModule(ctx context.Context) (rerr error) {
 	defer telemetry.End(span, func() error { return rerr })
 
 	resolveCtx, resolveSpan := Tracer().Start(ctx, "resolving module ref", telemetry.Encapsulate())
-	defer telemetry.End(resolveSpan, func() error { return rerr })
```

Comment:
Hmm I think this was fine before, but I can see how it's a little confusing.

Calling `End` multiple times on a span is fine - the later calls will be no-ops. The intent was to complete the span with either of the errors following the `getDefaultModuleConfiguration` too. So this pattern is a bit like having a `defer foo.Close()` as a safety net along with a `foo.Close()` in the happy path.

Reply:
Yeah, okay - I think that makes sense! I'll close this one then, I think it's unlikely that this one is actually causing any issues then.